packages/framework/src/Model/Cart/CartMigrationFacade.php

```diff
@@ -91,7 +91,7 @@ public function onKernelController(FilterControllerEvent $filterControllerEvent)
         $session = $filterControllerEvent->getRequest()->getSession();
 
         $previousCartIdentifier = $session->get(static::SESSION_PREVIOUS_CART_IDENTIFIER);
-        if (!empty($previousCartIdentifier) && $previousCartIdentifier !== $session->getId()) {
+        if ($previousCartIdentifier !== null && $previousCartIdentifier !== '' && $previousCartIdentifier !== $session->getId()) {
```

Comment:
I do not think so. The condition is for `$previousCartIdentifier` which may be anything.

Reply:
Yes, it can. 
But when it's `null` or `''`, it's inevitably different from `$session->getId()`, so the last condition should cover all previous scenarios, or am I mistaken?

Reply:
Maybe it should not but as I am not sure there is no cahnce of happening this I am moving more carefully.